nixos/modules/tasks/filesystems.nix

```diff
@@ -329,6 +331,18 @@ in
           options.
         '';
       }
+      {
+        assertion = config.boot.initrd.gptAutoRoot.enable -> config.boot.initrd.systemd.enable;
+        message = ''
+          'boot.initrd.gptAutoRoot.enable' requires 'boot.initrd.systemd.enable'.
+        '';
+      }
+      {
+        assertion = config.boot.initrd.gptAutoRoot.enable -> config.boot.loader.systemd-boot.enable;
+        message = ''
+          'boot.initrd.gptAutoRoot.enable' requires systemd-boot.
+        '';
+      }
```

Comment:
This is not the case if booting from a UKI with systemd-stub, where it will take care of setting up the EFI variables for the root disk: https://www.freedesktop.org/software/systemd/man/latest/systemd-stub.html#LoaderDevicePartUUID

Reply:
Well, we don't include ways of using systemd-stub at this time.

Reply:
Well, we have a builder for ukify? We maybe can't properly check if a UKI is used, but then this assertion should be removed

Reply:
Yeah I agree with @msanft we should not assert on that. Booting without enabling sd-boot via the module system is an actual use-case we have for appliance images.